### PR TITLE
Compliancy with tf-nightly

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

This PR makes the lib able to be used with the nightly builds of TensorFlow. And fix an issue with the min TensorFlow version.

### CODE REVIEW

File: src/transformers/file_utils.py [hunk 1]
```diff
@@ -89,10 +89,13 @@
...
-                _tf_version = None
-                _tf_available = False
+                try:
+                    _tf_version = importlib_metadata.version("tf-nightly")
+                except importlib_metadata.PackageNotFoundError:
+                    _tf_version = None
+                    _tf_available = False
     if _tf_available:
-        if version.parse(_tf_version) < version.parse("2"):
+        if version.parse(_tf_version) < version.parse("2.3"):
```
Comment (sgugger) [inline src/transformers/file_utils.py:98]: No this was just an old check for version 1 that was left, but we should not hardcode anything here, as we will forget to update it when we pin tensorflow to another release.

**Review by sgugger — APPROVED** [review]
I wonder how many different names tensorflow can have ;-) Thanks for fixing!

File: src/transformers/file_utils.py (hunk 1 above)
Comment (jplu) [reply src/transformers/file_utils.py:98]: Should I let it as it is or completely remove this condition?

Comment (sgugger) [reply src/transformers/file_utils.py:98]: I think we should leave it as is. Let's see what the others think!

Comment (LysandreJik) [reply src/transformers/file_utils.py:98]: I agree with Sylvain that this should stay as it is, it's to indicate that the general framework detected (TensorFlow 1) will not work correctly as we expect another (TensorFlow 2).

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Review by patrickvonplaten — APPROVED** [review]

**Comment by jplu** [comment]
Ok, just restored the previous version checking.

File: src/transformers/file_utils.py [hunk 2]
```diff
@@ -89,8 +89,20 @@
...
+                        _tf_version = importlib_metadata.version("tf-nightly")
+                    except importlib_metadata.PackageNotFoundError:
+                        try:
+                            _tf_version = importlib_metadata.version("tf-nightly-cpu")
+                        except importlib_metadata.PackageNotFoundError:
+                            try:
+                                _tf_version = importlib_metadata.version("tf-nightly-gpu")
+                            except importlib_metadata.PackageNotFoundError:
+                                _tf_version = None
+                                _tf_available = False
```
Comment (LysandreJik) [inline src/transformers/file_utils.py:105]: Wow!

Comment (jplu) [reply src/transformers/file_utils.py:105]: Yeah, just done a quick search on pypi to be sure we have all of them 😄

Comment (sgugger) [reply src/transformers/file_utils.py:105]: Might refactor this using a list of names at this stage ;-)

**Pull request merged**